tests/test-umockdev.py

```diff
@@ -179,6 +179,42 @@ def on_uevent(client, action, device, counters):
         mainloop.run()
         self.assertEqual(counter, [0, 0, 1, syspath, '0815', '1'])
 
+    def test_remove_usb_device(self):
+        '''testbed removing a single USB device'''
```

Comment:
This is too much duplication.  You can just append the `self.testbed.remove_device()` call to the existing `test_uevent()` above, run the main loop again, and check the counters. This is already set up for handling "remove" and "change" in the handler and counter.

Reply:
Done.